### DISCUSSION TITLE
Memory Integrity

### DISCUSSION THREAD

**Discussion by 26zl** [HotCakeX/Harden-Windows-Security#234] [Q&A] [answered]
Hi! First of all your work is amazing and its inspiring, thank you! 

I've had this problem for months now been trying to fix it, but the only thing that fixes it is not running the windows defender section of your script/program. But at the same time i want the beneficials of everything this script/program gives. 

So the problem is that memory integrity turns itself off, it will turn on first then when i restart it instantly turns off. I've tried to look in event viewer but cant find something that links with that problem. I have no driver problems. Memory integrity works when not runing the windows defender section of the script as stated. 

Both in regedit and gpedit says its on, but windows defender says otherwise. 
Here are some screenshots for reference!

Hope you can help! 

![Skjermbilde 2024-04-12 215352](https://github.com/HotCakeX/Harden-Windows-Security/assets/143036376/ee5f53b2-f32c-49b2-b9b3-a92bdd0edd5a)

![Skjermbilde 2024-04-12 215529](https://github.com/HotCakeX/Harden-Windows-Security/assets/143036376/9868a23e-f4a3-4eab-8d4c-349e08dec9b1)

![Skjermbilde 2024-04-12 215534](https://github.com/HotCakeX/Harden-Windows-Security/assets/143036376/226c74d4-7cc9-4b4e-ab8d-a84bb6960a31)

![Skjermbilde 2024-04-12 215541](https://github.com/HotCakeX/Harden-Windows-Security/assets/143036376/ab8fc9dd-d2e7-4be2-b4d2-3a573c65440d)

![Screenshot 2024-04-12 215820](https://github.com/HotCakeX/Harden-Windows-Security/assets/143036376/12935267-62b6-49dd-9a97-975e9f54c77d)

![Screenshot 2024-04-12 220401](https://github.com/HotCakeX/Harden-Windows-Security/assets/143036376/2e3acf6c-3f17-4872-947a-cf7b456b9816)

**Comment by HotCakeX** [comment, marked as answer]
Hi,
thank you very much and sorry for a bit late reply

I think i know the problem you're facing, it must be related to your hardware not supporting UEFI MAT

Can you try unchecking this box in group policy

<img width="506" alt="image" src="https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/3a7a0517-dc23-4a56-8ac0-b391e831e3f3">

then restart and activate Memory Integrity again, see if it stays turned on. I experienced the same issue on an old hardware.
Those Device Guard features are all turned on by Microsoft Security baselines category btw.

**Reply by 26zl** [reply]
Thank you so much! Haven't been able to answer because of some exams. But i tried it and has worked ever since. Thank you so much!

**Reply by HotCakeX** [reply]
@IllyrianCyber No worries, you're welcome ^^ glad to know that :)
